components/Participant.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { createRef, useEffect, useState } from 'react';
+
+const Participant = ({ gain, participant, isYou }) => {
+  const [videoRef] = useState(createRef<HTMLVideoElement>());
+
+  const playTrack = track => {
+    if (track.kind === 'audio') {
+      if (!isYou) {
+        // Don't play your own audio
+        var audio = new Audio();
+        audio.srcObject = postProcessAudioTrack(track);
+      }
+    } else if (track.kind === 'video') {
+      taggedLog('Creating new MediaStream');
+      taggedLog('track:', track);
+      const mediaStreamTrack = track.mediaStreamTrack;
+      const stream = new MediaStream();
+      stream.addTrack(mediaStreamTrack);
+      // @ts-ignore
+      videoRef.current.srcObject = stream;
+    } else {
+      taggedLog('non-audio or video track:', track);
+    }
+  };
+
+  useEffect(() => {
+    participant.tracks.forEach(publication => {
+      // sometimes it passes `null` (?!), so check if it's real
+      if (publication.track) {
+        playTrack(publication.track);
+      }
+    });
+
+    participant.on('trackSubscribed', track => {
+      taggedLog(`trackSubscribed (${track.kind})`, track);
+      playTrack(track);
+    });
+  }, [participant, videoRef.current]);
+
+  const postProcessAudioTrack = audioTrack => {
+    // Manually add the stream so that we can post-process the audio
+    // @ts-ignore
+    const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```

Comment:
I wonder if there's a polyfill (or if Babel has a plugin) for AudioContext/webkitAudioContext. I don't _love_ having to write/manage platform varied stuff like this :/

Reply:
Yeah, I agree. I think we can just write our own "polyfill":

```js
const AudioContext = window.AudioContext || window.webkitAudioContext;
```

In the future, I imagine us having a centralized `Audio` class that we do all this work from, rather than spreading it all over random React components like I did here. So the grossness will at least be centralized later on.